src/adapters/sudoswap-ethereum-1/index.ts

```diff
@@ -0,0 +1,387 @@
+            case "swapNFTsForAnyNFTsThroughERC20":
+                pairData = [
+                    new PairData(this.sdk, data.inputs[0][0], data.inputs[3], false, false),
+                    new PairData(this.sdk, data.inputs[0][1], data.inputs[3], true, false),
+                ];
+                break;
+            case "swapNFTsForAnyNFTsThroughETH":
+                pairData = [
+                    new PairData(this.sdk, data.inputs[0][0], data.inputs[3], false, false),
+                    new PairData(this.sdk, data.inputs[0][1], data.inputs[3], true, true),
+                ];
+                break;
+            case "swapNFTsForSpecificNFTsThroughERC20":
+                pairData = [
+                    new PairData(this.sdk, data.inputs[0][0], data.inputs[3], false, false),
+                    new PairData(this.sdk, data.inputs[0][1], data.inputs[3], true, false),
+                ];
+                break;
+            case "swapNFTsForSpecificNFTsThroughETH":
+                pairData = [
+                    new PairData(this.sdk, data.inputs[0][0], data.inputs[2], false, false),
+                    new PairData(this.sdk, data.inputs[0][1], data.inputs[2], true, false),
+                ];
+                break;
+            case "swapNFTsForToken":
+                pairData = [new PairData(this.sdk, [data.inputs[0]], data.inputs[2], false, false)];
+                break;
+        }
+
+        return pairData;
+    };
+
+    process = async (transaction: Transaction): Promise<ISaleEntity | undefined> => {
+        if (!transaction.blockNumber) return;
+
+        const block = await this.sdk.getBlock(transaction.blockNumber);
+        const timestamp = moment.unix(block.timestamp as number).utc();
+
+        const pairData = await this._decodeData(transaction.input);
+
+        if (!pairData) return;
+
+        for (const pair of pairData) {
+            for (const pairInfo of pair.info) {
+                const { price, priceUsd } = await this._getPrice(
+                    pairInfo.pairAddress,
+                    pairInfo.amount,
+                    pairInfo.isBuyCall,
+                    block,
+                );
+
+                console.log(`Price: ${price}, pair: ${pairInfo.pairAddress}, NftId: ${pairInfo.nfts}`);
+                console.log(`transaction Hash: ${transaction.hash}`);
+
+                const entity: ISaleEntity = {
+                    providerName: this.name,
+                    providerContract: this.contract.toLowerCase(),
+                    protocol: this.protocol,
+                    nftContract: pairInfo.pairAddress.toLowerCase(),
+                    nfts: pairInfo.nfts,
+                    token: this.token.toLowerCase(),
+                    tokenSymbol: this.symbol?.symbol || "",
+                    price: price,
+                    priceUsd: priceUsd,
+                    seller: pairInfo.pairAddress.toLowerCase(),
+                    buyer: pairInfo.buyer.toLowerCase(),
```

Comment:
Would you mind checking if `seller` and `buyer` properties are correct?
I've tested it and it returns buyer as a seller and vice versa.

Reply:
That's because in `Sudoswap` a user can buy or sell an NFT for ETH. So if a user is buying an NFT then the user address is the buyer otherwise if a user is selling an NFT then the user address is the seller.

Reply:
Buyer and seller are mixed for [this transaction](https://etherscan.io/tx/0x0c4255293c80b95fb972613b87171eccb7b532ab3fe95cbb17f7136d218965ea). You can run a tester and check if the `buyer`/`seller` is ok in the entities.

<img width="694" alt="image" src="https://user-images.githubusercontent.com/82936197/192225999-f54e04a9-f472-4889-9ed5-9d8854d42307.png">

Reply:
@liutaurasjanuskas in this [transaction](https://etherscan.io/tx/0x0c4255293c80b95fb972613b87171eccb7b532ab3fe95cbb17f7136d218965ea) the user is exchanging multiple NFT's with NFT's and token (function call: `robustSwapETHForSpecificNFTsAndNFTsToToken`) and in this particular entity the user is selling the [Tubby Cats (TUBBY)](https://etherscan.io/token/0xca7ca7bcc765f77339be2d648ba53ce9c8a262bd) NFT (id: `19345`) to the contract. So the seller in this particular entity is correct.

Reply:
@harsh9200, gotcha, thanks. Is there an easy way to ignore other sales rather than ERC721/ERC1155 to ERC20? Currently we have no architecture to store NFT to NFT trades, but we want to finish this integration.

Reply:
yes, we can probably do that by commenting out the function calls for NFT to NFT type trades. Do you want me to add that to this PR?

Reply:
Yes, do it please.

Reply:
Done please check